pkg/helm/helm.go

```diff
@@ -86,17 +88,17 @@ func UpdateIndex(original, new *helmRepo.IndexFile) (*helmRepo.IndexFile, bool)
 			// Try to preserve it only if nothing has changed.
 			if originalChartVersion.Digest == chartVersion.Digest {
 				// Don't modify created timestamp
-				updatedIndex.Entries[chartName][i].Created = originalChartVersion.Created
+				newRepo.Entries[chartName][i].Created = originalChartVersion.Created
 			} else {
 				upToDate = false
 				logrus.Debugf("Chart %s at version %s has been modified", chartName, originalChartVersion.Version)
 			}
 		}
 	}
 
-	for chartName, chartVersions := range original.Entries {
+	for chartName, chartVersions := range origRepo.Entries {
 		for _, chartVersion := range chartVersions {
-			if !updatedIndex.Has(chartName, chartVersion.Version) {
+			if !newRepo.Has(chartName, chartVersion.Version) {
```

Comment:
This still relies on the underling *helmrepo.Indexfile `semver` so the check of index.yaml being up to date doesn't work in the case where two versions `103.0.1` having different upstream tags